### PR TITLE
ExprExplodedBlocks - fix not being able to remove

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to fix an issue with not being able to remove blocks from the exploded block list.
This is standard behaviour in Bukkit, but Skript "lists" don't allow for this action, which I've seen many many people complain about.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

**Review by Moderocky — APPROVED** [review]

**Review by Fusezion — APPROVED** [review]
Wanna say I was suppose to be fixed in #3990 but like much of those prs from then it staled out all for this fix.

File: src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java [hunk 1]
```diff
@@ -68,6 +75,25 @@ protected Block[] get(Event e) {
 		List<Block> blockList = ((EntityExplodeEvent) e).blockList();
 		return blockList.toArray(new Block[blockList.size()]);
 	}
+
+	@Override
+	public @Nullable Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.REMOVE)
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java:81]: Are there other operations we could support? At least `DELETE` would work, as in `clear the exploded blocks`?

Comment (Fusezion) [reply src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java:81]: actually would be good to support the full set of `SET`, `ADD`, `REMOVE`, `DELETE`

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java:81]: oh yeah that would be a good idea. Ill whip that up and test it

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java [hunk 2]
```diff
@@ -68,6 +81,42 @@ protected Block[] get(Event e) {
...
+			return;
+		}
+		if (mode == ChangeMode.SET)
+			blocks.clear();
+		for (Object object : delta) {
+			if (object instanceof Block) {
+				if (mode == ChangeMode.REMOVE)
+					blocks.remove((Block) object);
+				else if (mode == ChangeMode.ADD || mode == ChangeMode.SET)
+					blocks.add((Block) object);
```
[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java]: all 3 suggestions applied 👍🏻

**Review by sovdeeth — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java [hunk 3]
```diff
@@ -68,6 +85,45 @@ protected Block[] get(Event e) {
...
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (!(event instanceof EntityExplodeEvent))
+			return;
+
+		List<Block> blocks = ((EntityExplodeEvent) event).blockList();
+		if (mode == ChangeMode.DELETE) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java:109]: i think a switch statement would be better here

File: src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java [hunk 4]
```diff
@@ -68,6 +85,45 @@ protected Block[] get(Event e) {
...
+
+	@Override
+	public @Nullable Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case REMOVE:
+			case SET:
+				return CollectionUtils.array(Block[].class);
+			case DELETE:
+				return CollectionUtils.array();
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java:97]: it actually should be okay to return an array with contents for DELETE. i don't believe EffChange cares about the contents for DELETE/RESET, just that it isn't null

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
i wonder if this might be better to target feature? does attempting to modify the list (remove, add, etc.) parse currently?

**Comment by ShaneBeee** [comment]
> i wonder if this might be better to target feature? does attempting to modify the list (remove, add, etc.) parse currently?

It doesn't appear to error (which is odd) but absolutely nothing happens.

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**